crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -171,6 +173,8 @@ pub struct CumulocityConverter {
     pub c8y_endpoint: C8yEndPoint,
     pub mqtt_schema: MqttSchema,
     pub entity_store: EntityStore,
+    pub external_id_map: HashMap<EntityTopicId, String>,
+    pub reverse_external_id_map: HashMap<String, EntityTopicId>,
```

Comment:
@didier-wenzek and I have agreed to avoid the duplication of these maps here and reuse the mapping in the entity store itself for the same.

The key thing is that, the mapping of `EntityTopicId` to its C8y representation would be done at the converter level, as it is a cloud specific logic, but the mapped id will be used inside the entity store as the `id` of the entity. From the entity store's perspective, it's just one unique opaque ID.  That way, we have clear separation of concerns. So, when different kinds of mappers/components use the `EntityStore` data structure, they are free to do different kinds of `id` mappings and store those mapped ids in the same struct, without needing yet another external mapping.